server/src/services/validation/analyse.ts

```diff
@@ -16,6 +16,11 @@ export async function analyse(
     const solFileEntry = getOrInitialiseSolFileEntry(serverState, internalUri);
 
     await analyzeSolFile(serverState, solFileEntry, changeDoc.getText());
+
+    // Notify that a file was successfully analyzed
+    serverState.connection.sendNotification("custom/analyzed", {
```

Comment:
Is this to support testing?

Reply:
Yes. When a file is opened, a notification is sent to the server but it doesn't wait for a response, so this is in place to let the client know that we finished analyzing  an open file. Still I think in the future we should delegate this to the server, and have it wait on each of the requests that require a file to be analyzed. I.e. on a definition request, if the file is not analyzed yet, have that request wait for it instead of rejecting it. This would remove the need for this custom notification.

Reply:
Can we suppress this if the env is not testing then?

I really like the trigger analysis if it is not already started approach. I will add a linear ticket for that.

Reply:
Sure, will push the change now!